### ISSUE TITLE
Add a rule prefer-exact-props

### ISSUE THREAD

**Issue by julienw** [yannickcr/eslint-plugin-react#1455]
With react and flow, it's very common to type props like this:

```javascript
type Props = {
  prop1: string,
  prop2: number
};
```

However this makes it possible to pass other props as well, as by default objects in Flow are not exact. So this can hide an issue where the developer removed a prop but the caller still use it:
```javascript
<Component prop1='foo' prop2={42} prop3='bar' />
```
Or made a typo (this example is possible is `prop2` is optional):
```javascript
<Component prop1='foo' porp2={42} />
```

To fix this with flow, the solution is to use exact objects:
```javascript
type Props = {|
  prop1: string,
  prop2: number
|};
```

It would be useful to have an eslint rule to force this pattern.

**Comment by ljharb** [comment]
I'm a bit hesitant to have Flow-only rules in a non-flow-specific eslint plugin, but I agree that such a rule would be a good thing to have if using Flow.

With PropTypes, you can use https://npmjs.com/prop-types-exact to get the same behavior; maybe the rule could recommend both?

**Comment by julienw** [comment]
Ah yep, good idea !
